repositories/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala

```diff
@@ -72,19 +72,27 @@ object Commit {
   implicit val commitFormat: OFormat[Commit] = Json.format[Commit]
 }
 
-final case class ProjectFile(path: Path, content: String)
-
 object ProjectFile {
-  implicit object ProjectFileFormat extends Format[ProjectFile] {
-    override def reads(json: JsValue): JsResult[ProjectFile] =
-      JsSuccess(ProjectFile(Paths.get((json \ "path").as[String]), (json \ "content").as[String]))
 
-    override def writes(o: ProjectFile): JsValue = JsObject(
-      Seq("path" -> JsString(o.path.toString), "content" -> JsString(o.content))
-    )
+  implicit object pathFormat extends Format[Path] {
+    override def writes(o: Path): JsValue = JsString(o.toString)
+    override def reads(json: JsValue): JsResult[Path] = JsSuccess(Paths.get(json.as[String]))
```

Comment:
This line is equivalent to something like this.
```
def parseLong(longStr: String): Try[Long] = Success(longStr.toLong)
```
I assume you understand why this is bad.
Use `json.validate` instead of `json.as` and do not wrap it manually into a `JsSuccess`.

Reply:
json.validate needs formatter itself, method validate needs implicit parameter - formatter to work

Reply:
So what? Take a look at the implementation of the `.as` method. If you can use `.as` method, then you can also use `.validate` method without a problem.

Reply:
agreed, ofcourse, done